public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -461,9 +461,9 @@ export class LokiDatasource
    * Retrieve the current time range as Loki parameters.
```

Comment:
We should update the method description. But, to be honest, I think what this is doing is not a concern for the datasource class. Maybe a function or something from `grafana-data` that we can use?

Reply:
I do have a recommendation. It currently says `Retrieve the current time range as Loki parameters.`, but it now receives an argument.

We should reflect that in the description as something like `Given a time range, returns it as Loki parameters`.

And in that sense, what do you think about making the `timeRange` required and not optional? Because this method is actually doing two things:
1) No argument: call `this.timeRange()` and return it as Loki parameters.
2) With argument: return it as Loki parameters.

If we make it mandatory, it will only do one thing: grab a timeRange (any) and return it as Loki parameters.

The current calls without argument should do `getTimeRangeParams(this.getTimeRange())`.

Reply:
> And in that sense, what do you think about making the timeRange required and not optional?

Yes, that is a goal, but in this PR I am focusing only on changes in Language provider. In the next PR, the goal is to focus on `datasource.ts` and tacke also this - as otherwise it is quite an owerwhelming change.

Reply:
Updating comment to `Given a time range, returns it as Loki parameters`

Reply:
Updated in https://github.com/grafana/grafana/pull/78450/commits/f6b762a95452fd5601684c854bd41261e6031ddd